cmd/substreams/init.go

```diff
@@ -172,8 +172,63 @@ func runSubstreamsInitE(cmd *cobra.Command, args []string) error {
 			return fmt.Errorf("failed to call discovery endpoint: %w", err)
 		}
 
-		var options []huh.Option[*pbconvo.DiscoveryResponse_Generator]
+		type BlockchainProtocolSelector struct {
+			Id         string
+			Title      string
+			Generators []*pbconvo.DiscoveryResponse_Generator
+		}
+		titleFromGroup := func(group string) string {
+			switch group {
+			case "evm":
+				return "EVM"
+			case "solana":
+				return "Solana"
+			case "cosmos":
+				return "Cosmos"
+			case "starknet":
+				return "Starknet"
+			}
+
+			return ""
+		}
+
+		var filteredProtocols map[string]*BlockchainProtocolSelector = make(map[string]*BlockchainProtocolSelector)
 		for _, gen := range resp.Msg.Generators {
+			selector, groupExists := filteredProtocols[gen.Group]
+			if groupExists {
+				selector.Generators = append(selector.Generators, gen)
+			} else {
+				filteredProtocols[gen.Group] = &BlockchainProtocolSelector{
+					Id:         gen.Group,
+					Title:      titleFromGroup(gen.Group),
+					Generators: []*pbconvo.DiscoveryResponse_Generator{gen},
+				}
+			}
+		}
+
+		var protocolOptions []huh.Option[*BlockchainProtocolSelector]
+		for _, value := range filteredProtocols {
+			protocolOptions = append(protocolOptions, huh.Option[*BlockchainProtocolSelector]{
+				Key:   value.Title,
+				Value: value,
+			})
+		}
+
+		var selector *BlockchainProtocolSelector
+		selectField := huh.NewSelect[*BlockchainProtocolSelector]().
+			Title("Choose the protocol you inted to work with:").
+			Options(protocolOptions...).
+			Value(&selector)
+
+		err = huh.NewForm(huh.NewGroup(selectField)).WithTheme(huhTheme).WithAccessible(WITH_ACCESSIBLE).Run()
+		if err != nil {
+			return fmt.Errorf("failed taking input: %w", err)
+		}
+
+		fmt.Println(gray("┃"), "Chosen chain: ", bold(selector.Id), "-", selector.Title)
```

Comment:
You've used `protocol` in the question but `Chosen chain` here, shouldn't it be `Chosen protocol: `?